src/nvapi_d3d12.cpp

```diff
@@ -203,6 +207,8 @@ extern "C" {
         if (adapter == nullptr || (!adapter->HasNvProprietaryDriver() && !adapter->HasNvkDriver()))
             return Ok(str::format(n, " (sm_0)"));
 
+        pGraphicsCaps->bVariablePixelRateShadingSupported = adapter->IsVariablePixelRateShadingSupported();
```

Comment:
In the incredibly unlikely scenario that we are running this code alongside some _ancient_ version of vkd3d-proton (or usage of the ext is disabled via `VKD3D_DISABLE_EXTENSIONS`), the safer choice would be to check which extensions were actually enabled on the logical device instead. That would look something like

```cpp
Com<ID3D12DXVKInteropDevice> device;
if (SUCCEEDED(pDevice->QueryInterface(IID_PPV_ARGS(&device)))) {
    std::vector<const char*> extensions;
    UINT count;
    if (SUCCEEDED(device->GetDeviceExtensions(&count, nullptr))) {
        extensions.resize(count);
        if (SUCCEEDED(device->GetDeviceExtensions(&count, extensions.data())))
            pGraphicsCaps->bVariablePixelRateShadingSupported = std::any_of(extensions.begin(), extensions.end(), [](auto ext) { return ext == std::string_view{VK_KHR_FRAGMENT_SHADING_RATE_EXTENSION_NAME}; });
    }
}
```

but it's fairly disgusting and probably overcomplicated for such a rare corner-case, so if you are fine with this small risk then let's just leave a comment that would explain potential mismatch and ignore the problem for now.